shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeIngressServiceImpl.java

```diff
@@ -83,13 +85,14 @@ public void createBridgeIngress(BridgeDTO bridgeDTO) {
 
     @Override
     public void deleteBridgeIngress(BridgeDTO bridgeDTO) {
-        final String namespace = customerNamespaceProvider.resolveName(bridgeDTO.getCustomerId());
-        final boolean bridgeDeleted =
-                kubernetesClient
-                        .resources(BridgeIngress.class)
-                        .inNamespace(namespace)
-                        .delete(BridgeIngress.fromDTO(bridgeDTO, namespace));
-        if (!bridgeDeleted) {
+        String customerId = bridgeDTO.getCustomerId();
+        String bridgeId = bridgeDTO.getId();
+
+        // Deletion of the Namespace will cascade deletion of the BridgeIngress
+        final Namespace namespace = customerBridgeNamespaceProvider.fetchOrCreateCustomerBridgeNamespace(customerId, bridgeId);
```

Comment:
Since we have a `Namespace` per `Bridge` we can delete the whole `Namespace`.

I discussed cascade deletion with the Java Operator SDK team and they confirmed it's behaviour.